dev/com.ibm.ws.microprofile.config.1.3_fat/fat/src/com/ibm/ws/microprofile/config13/test/VariableServerXMLTest.java

```diff
@@ -0,0 +1,155 @@
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.topology.impl.LibertyServer;
+import componenttest.topology.utils.FATServletClient;
+
+/**
+ * Example Shrinkwrap FAT project:
+ * <li> Application packaging is done in the @BeforeClass, instead of ant scripting.
+ * <li> Injects servers via @Server annotation. Annotation value corresponds to the
+ * server directory name in 'publish/servers/%annotation_value%' where ports get
+ * assigned to the LibertyServer instance when the 'testports.properties' does not
+ * get used.
+ * <li> Specifies an @RunWith(FATRunner.class) annotation. Traditionally this has been
+ * added to bytecode automatically by ant.
+ * <li> Uses the @TestServlet annotation to define test servlets. Notice that no @Test
+ * methods are defined in this class. All of the @Test methods are defined on the test
+ * servlet referenced by the annotation, and will be run whenever this test class runs.
+ */
+@RunWith(FATRunner.class)
+public class VariableServerXMLTest extends FATServletClient {
+
+    public static final String APP_NAME = "variableServerXMLApp";
+
+    @Server("serverXMLVariableServer")
+    @TestServlet(servlet = VariableServerXMLServlet.class, contextRoot = APP_NAME)
+    public static LibertyServer server;
+
+    @BeforeClass
+    public static void setUp() throws Exception {
+        // Create a WebArchive that will have the file name 'app1.war' once it's written to a file
+        // Include the 'app1.web' package and all of it's java classes and sub-packages
+        // Automatically includes resources under 'test-applications/APP_NAME/resources/' folder
+        // Exports the resulting application to the ${server.config.dir}/apps/ directory
+        ShrinkHelper.defaultApp(server, APP_NAME, "com.ibm.ws.microprofile.config13.variableServerXML.*");
+
+        server.startServer();
+    }
+
+    @Test
+    public void testBasicConfig() throws Exception {
+        test(server, "/variableServerXMLApp/ServerXMLVariableServlet?testMethod=varPropertiesBaseTest");
+    }
+
+    @Test
+    public void testConfigOrder() throws Exception {
+        test(server, "/variableServerXMLApp/ServerXMLVariableServlet?testMethod=varPropertiesOrderTest");
+    }
+
+    @Test
+    public void testChangeVariablesConfig() throws Exception {
+
+        // run the "before" test to check the value of the variable before the server.xml is updated
+        test(server, "/variableServerXMLApp/ServerXMLVariableServlet?testMethod=varPropertiesBeforeTest");
+
+        // switch to new configuration
+        server.copyFileToLibertyServerRoot("refreshVariables/server.xml");
+
+        // Wait for message: "the server configuration was successfully updated"
+        assertNotNull("The server.xml was not updated",
+                      server.waitForStringInLog("CWWKG0017I"));
+        Thread.sleep(1000); // We need this pause so that the config change is picked up through the polling mechanism,
```

Comment:
if it takes time for the change in server.xml to be detected, shouldn't this wait be before we check for the string in the log?

Reply:
The waitForStringInLog() method waits for the server.xml to be refreshed and for the appropriate message to appear in the log. The sleep allows time for a delay in the mp config polling mechanism re-reading that (new) configuration.